src/executor/nested_loop_join_executor.cpp

```diff
@@ -74,26 +76,23 @@ bool NestedLoopJoinExecutor::DInit() {
  * to join the left and right result.
  *
  */
+
+// For this version, the work flow is that we first lookup the left table, and
+// use the result to lookup right table. If left table is done that means right
+// table is also done. So we only keep the left_child_done_ as the sign.
 bool NestedLoopJoinExecutor::DExecute() {
```

Comment:
Yes, since UpdatePredicate can be overloaded by each Executor including SeqScan. But I plan to implement index predicate with seq predicate together, so for now it doesn't work. I will implement it in the next version (with index predicate).